Bear is called on each build

karkhaz opened the pull request:
The bear tool is now used to wrap each invocation of makepkg when
attempting to build packages. The output is stored as part of the
JSON result structure.

karkhaz committed:
Allow user to specify toolchain to post-process

The user can now optionally specify a list of toolchains to
post-process, instead of post-processing the results of all toolchains
(which remains the default). This is because post-processing it taking a
large amount of main memory, and running the post-processing one
toolchain at a time can help the machine to not thrash.

karkhaz committed:
More accurate build count on HTML summary pages

Summary pages can display the total number of builds on that page.
Previous to this commit, summary pages that displayed builds for more
than one toolchain would count each build once for each toolchain, i.e.
tripling the apparent number of builds if there are three toolchains on
display.

This commit changes that behaviour so that the "total" number of builds
is the number of builds divided by the number of toolchains on display.

karkhaz committed:
Improved memory efficiency of blocker post-process

Post-processed results are no longer stored in an inter-process data
structure while waiting to have their blockers calculated, as this was a
huge waste of memory. Instead, results are written to disk after all
post-processing except for blocker analysis is done. When all results
have been written, only the necessary data for blocker analysis is re-
loaded from disk; the on-disk results are then updated with the blocker
results.

Due to this improvement, the blocker analysis has been re-enabled by
default.

petrhosek commented on bear/PKGBUILD:
```diff
@@ -0,0 +1,33 @@
+md5sums=('SKIP')
+
+build() {
+	cd "$srcdir"/bear
```
The indentation in this file seems to be broken.

petrhosek commented on bear/PKGBUILD:
```diff
@@ -0,0 +1,33 @@
+pkgver=2.1.5
+pkgrel=1
+pkgdesc="tool to generate compilation database for clang tooling"
+arch=('i686' 'x86_64')
```
Does it mean bear doesn't work on arm? Have you tried it?

karkhaz replied:
These are the only two possible values in the `arch` array of a PKGBUILD. ARM is not supported on Arch Linux.

karkhaz committed:
Bear is called on each build

The bear tool, which has been added as a submodule of this project, is
used to wrap each invocation of makepkg. The output is stored as part of
the JSON result structure.
